tools/pd-api-bench/cases/controller.go

```diff
@@ -0,0 +1,268 @@
+		gRPCClients: gRPCClients,
+		http:        make(map[string]*httpController),
+		grpc:        make(map[string]*gRPCController),
+	}
+}
+
+// GetHTTPCase returns the HTTP case config.
+func (c *Coordinator) GetHTTPCase(name string) (*Config, error) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if controller, ok := c.http[name]; ok {
+		return controller.GetConfig(), nil
+	}
+	return nil, errors.Errorf("case %v does not exist.", name)
+}
+
+// GetGRPCCase returns the gRPC case config.
+func (c *Coordinator) GetGRPCCase(name string) (*Config, error) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if controller, ok := c.grpc[name]; ok {
+		return controller.GetConfig(), nil
+	}
+	return nil, errors.Errorf("case %v does not exist.", name)
+}
+
+// GetAllHTTPCases returns the all HTTP case configs.
+func (c *Coordinator) GetAllHTTPCases() map[string]*Config {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	ret := make(map[string]*Config)
+	for name, c := range c.http {
+		ret[name] = c.GetConfig()
+	}
+	return ret
+}
+
+// GetAllGRPCCases returns the all gRPC case configs.
+func (c *Coordinator) GetAllGRPCCases() map[string]*Config {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	ret := make(map[string]*Config)
+	for name, c := range c.grpc {
+		ret[name] = c.GetConfig()
+	}
+	return ret
+}
+
+// SetHTTPCase sets the config for the specific case.
+func (c *Coordinator) SetHTTPCase(name string, cfg *Config) error {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if fn, ok := HTTPCaseFnMap[name]; ok {
+		var controller *httpController
+		if controller, ok = c.http[name]; !ok {
+			controller = newHTTPController(c.ctx, c.httpClients, fn)
+			c.http[name] = controller
+		}
+		controller.stop()
+		controller.SetQPS(cfg.QPS)
+		if cfg.Burst > 0 {
+			controller.SetBurst(cfg.Burst)
+		}
+		controller.run()
+	} else {
+		return errors.Errorf("HTTP case %s not implemented", name)
+	}
+	return nil
+}
+
+// SetGRPCCase sets the config for the specific case.
+func (c *Coordinator) SetGRPCCase(name string, cfg *Config) error {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if fn, ok := GRPCCaseFnMap[name]; ok {
+		var controller *gRPCController
+		if controller, ok = c.grpc[name]; !ok {
+			controller = newGRPCController(c.ctx, c.gRPCClients, fn)
+			c.grpc[name] = controller
+		}
+		controller.stop()
+		controller.SetQPS(cfg.QPS)
+		if cfg.Burst > 0 {
+			controller.SetBurst(cfg.Burst)
+		}
+		controller.run()
+	} else {
+		return errors.Errorf("HTTP case %s not implemented", name)
+	}
+	return nil
+}
+
+type httpController struct {
+	HTTPCase
+	clients []pdHttp.Client
+	pctx    context.Context
+
+	ctx    context.Context
+	cancel context.CancelFunc
+	wg     sync.WaitGroup
+}
+
+func newHTTPController(ctx context.Context, clis []pdHttp.Client, fn HTTPCraeteFn) *httpController {
+	c := &httpController{
+		pctx:     ctx,
+		clients:  clis,
+		HTTPCase: fn(),
+	}
+	return c
+}
+
+// run tries to run the HTTP api bench.
+func (c *httpController) run() {
+	if c.GetQPS() <= 0 || c.cancel != nil {
```

Comment:
How about also validating it when setting it?

Reply:
This is a lazy implementation. In this way, the operation of adding, modifying, and deleting an API is implemented.